eng/intellisense.targets

```
@@ -0,0 +1,41 @@
+<Project>
+
+  <PropertyGroup>
+    <XmlDocFileRoot>$([MSBuild]::NormalizeDirectory('$(NuGetPackageRoot)', 'microsoft.private.intellisense', '$(MicrosoftPrivateIntellisenseVersion)', 'IntellisenseFiles'))</XmlDocFileRoot>
+  </PropertyGroup>
+
+  <!-- If the project is not generating its own documentation yet, retrieve it from the dotnet-api-docs nuget package. -->
+  <Choose>
+    <When Condition="('$(UseIntellisenseDocumentationFile)' == '' or '$(UseIntellisenseDocumentationFile)' == 'false') and Exists($([MSBuild]::NormalizePath('$(XmlDocFileRoot)', 'net', '1033', '$(AssemblyName).xml')))">
+      <PropertyGroup>
+        <IntellisenseFilePath>$([MSBuild]::NormalizePath('$(XmlDocFileRoot)', 'net', '1033', '$(AssemblyName).xml')</IntellisenseFilePath>
+      </PropertyGroup>
+    </When>
+    <When Condition="('$(UseIntellisenseDocumentationFile)' == '' or '$(UseIntellisenseDocumentationFile)' == 'false') and Exists($([MSBuild]::NormalizePath('$(XmlDocFileRoot)', 'dotnet-plat-ext', '1033', '$(AssemblyName).xml')))">
+      <PropertyGroup>
+        <IntellisenseFilePath Condition="">$([MSBuild]::NormalizePath('$(XmlDocFileRoot)', 'dotnet-plat-ext', '1033', '$(AssemblyName).xml')</IntellisenseFilePath>
+      </PropertyGroup>
+    </When>
+  </Choose>
+
+  <!-- TODO: Remove this target when no library relies on the intellisense documentation file anymore.-->
+  <Target Name="ChangeDocumentationFileForPackaging"
+          AfterTargets="DocumentationProjectOutputGroup"
+          Condition="'$(UseIntellisenseDocumentationFile)' == 'true'">
+    <ItemGroup>
+      <DocumentationProjectOutputGroupOutput Remove="@(DocumentationProjectOutputGroupOutput)" />
+      <DocumentationProjectOutputGroupOutput Include="$(IntellisenseFilePath)" />
+    </ItemGroup>
+  </Target>
+
+  <Target Name="CopyDocumentationFileToXmlDocDir"
+          AfterTargets="CoreCompile"
```

IIRC, @ericstj was mentioning that the `AfterTargets` property here shouldn't be `CoreCompile`, since the `DocumentationProjectOutputGroupOutput` is empty.

---

Not necessarily because `DocumentationProjectOutputGroupOutput`.  That will get set because this depends on `ChangeDocumentationFileForPackaging` which in turn depends on `DocumentationProjectOutputGroup` which sets `DocumentationProjectOutputGroupOutput`.  The problem is that the file isn't yet in the final location after `CoreCompile`, which is why I made [this comment](https://github.com/dotnet/runtime/pull/79134/files#r1067565007).  As I mentioned there, you should make this `AfterTargets=CopyFilesToOutputDirectory`.
```suggestion
          AfterTargets="CopyFilesToOutputDirectory"
```